odl/tomo/geometry/geometry.py

```diff
@@ -0,0 +1,383 @@
+
+        if not isinstance(detector, Detector):
+            raise ValueError('motion_grid {!r} not a `Detector` instance.'
+                             ''.format(detector))
+
+        self._ndim = int(ndim)
+        self._motion_grid = motion_grid
+        self._motion_params = motion_grid.convex_hull()
+        self._detector = detector
+        self._implementation_cache = {}
+
+    @property
+    def motion_params(self):
+        """The motion parameters given as an `IntervalProd`."""
+        return self._motion_params
+
+    @property
+    def motion_grid(self):
+        """A sampling grid for `motion_params`."""
+        return self._motion_grid
+
+    @property
+    def det_params(self):
+        """The detector parameters."""
+        return self.detector.params
+
+    @property
+    def det_grid(self):
+        """A sampling grid for `det_params`."""
+        return self.detector.param_grid
+
+    @property
+    def params(self):
+        """Joined motion and detector parameters.
+
+        Returns an `IntervalProduct` with the motion parameters inserted
+        before the detector parameters.
+        """
+        return self.motion_params.append(self.det_params)
+
+    @property
+    def grid(self):
+        """Joined sampling grid for motion and detector parameters.
+
+        The sampling grid for detector parameters is appended to the
+        sampling grid for the motion parameters.
+        """
+        return self.motion_grid.append(self.det_grid)
+
+    @property
+    def detector(self):
+        """The detector representation."""
+        return self._detector
+
+    @abstractmethod
+    def det_refpoint(self, mpar):
+        """The detector reference point function.
+
+        Parameters
+        ----------
+        mpar : element of motion parameters
+            Motion parameter for which to calculate the detector
+            reference point
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+            The reference point, an `ndim`-dimensional vector
+        """
+
+    @abstractmethod
+    def rotation_matrix(self, mpar):
+        """The detector rotation function for calculating the detector
+        reference position.
+
+        Parameters
+        ----------
+        mpar : element of motion parameters `motion_params`
+            Motion parameter for which to calculate the detector
+            reference rotation
+
+        Returns
+        -------
+        rot : `numpy.ndarray`, shape (`ndim`, `ndim`)
+            The rotation matrix mapping the standard basis vectors in
+            the fixed ("lab") coordinate system to the basis vectors of
+            the local coordinate system of the detector reference point,
+            expressed in the fixed system.
+        """
+
+    def det_to_src(self, mpar, dpar, normalized=True):
+        """Vector pointing from a detector location to the source.
+
+        A function of the motion and detector parameters.
```

Comment:
This line is obvious, perhaps more text on what this means in some "usual" cases so users can relate to it?